### PR TITLE
76 Optimize index finding in scispacy processor

### PR DESCRIPTION
**Pull request by nikhilranjan7**
This PR fixes 76. 

### Description of changes
Describe what are the changes, and how they solve the problem.
Change token span index to char index

### Possible influences of this PR.
Describe what are the possible side-effects of the code change.
Hyponym detection more simplified indexing

### Test Conducted
Describe what test cases are included for the PR.
Scispacy processor unittest

### CODE REVIEW

File: fortex/health/processors/scispacy_processor.py [hunk 1]
```diff
@@ -93,36 +93,24 @@ def _process(self, input_pack: DataPack):
             else:
```
Comment (bhaskar2443053) [inline fortex/health/processors/scispacy_processor.py:93]: we should have a check for self.configs.pipe_name == "hyponym_detector" too.
and through an error message if its not from the options in config.

File: tests/fortex/health/processors/scispacy_processor_test.py [hunk 2]
```diff
@@ -83,8 +83,8 @@ def test_ScispaCy_Hypomym_processor(self):
         pack = self.nlp.process(document)
         expected_value = {
             "hyponym_link": "such_as",
-            "general": "Keystone plant species",
-            "specific": "fig trees",
+            "specific": "Keystone plant species",
```
Comment (Piyush13y) [inline tests/fortex/health/processors/scispacy_processor_test.py:86]: Shouldn't it be the other way around, like it was before these changes?

Comment (nikhilranjan7) [reply tests/fortex/health/processors/scispacy_processor_test.py:86]: general is hypernym and specific is hyponym
image

Comment (Piyush13y) [reply tests/fortex/health/processors/scispacy_processor_test.py:86]: That's fine, however, what I meant to say was "Keystone plant species" is the general concept while "fig trees" is the specific concept and not the other way around. You can refer https://pythonlang.dev/repo/allenai-scispacy/ to check the ordering on concepts in the returned tuple in Hearst patterns. item[1] should be the general concept and item[2] should be the specific concept, and we should label accordingly. Currently, we are doing the exact opposite I think. Let me know if I have misunderstood.

**Comment by Piyush13y** [comment]
I was wondering how the scispacy test case passed when you had swapped the general and specific concepts in the backend initially (with the 1st commit). Could you check that out, since if the tests are running properly, it should have failed when we swapped both the concepts, right?

**Pull request merged**